### PR TITLE
WebsocketProviderOptions has been adjusted

### PR DESCRIPTION
**Pull request by AsyncLegs**
requestOptions field has added: more info - https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/websocket/index.d.ts#L697

### CODE REVIEW

**Review by nivida — CHANGES_REQUESTED** [review]
The types you were editing are for the 2.x architecture but the [PR](https://github.com/ethereum/web3.js/pull/2979) you opened with the ``requestOptions`` addition is for the 1.0 architecture. The ``requestOptions`` can be added to the [createWebsocketProvider](https://github.com/ethereum/web3.js/blob/2.x/packages/web3-providers/src/factories/ProvidersModuleFactory.js#L136) method of the [ProvidersModuleFactory](https://github.com/ethereum/web3.js/blob/2.x/packages/web3-providers/src/factories/ProvidersModuleFactory.js#L136) if you would like to add this addition also in 2.0.

**Comment by AsyncLegs** [comment]
Oh sorry, just missed a bit, because a reconnect [method](https://github.com/ethereum/web3.js/blob/2.x/packages/web3-providers/src/providers/WebsocketProvider.js#L100)  has this option already.
But nevertheless, PR has been updated.

File: packages/web3-providers/src/factories/ProvidersModuleFactory.js [hunk 1]
```diff
@@ -126,14 +126,15 @@ export default class ProvidersModuleFactory {
         // runtime is of type node
         if (typeof process !== 'undefined' && process.versions != null && process.versions.node != null) {
             let headers = options.headers || {};
+            const requestOptions = options.requestOptions || null;
```
Comment (nivida) [inline packages/web3-providers/src/factories/ProvidersModuleFactory.js]:
```suggestion

```

File: packages/web3-providers/src/factories/ProvidersModuleFactory.js [hunk 2]
```diff
@@ -126,14 +126,15 @@ export default class ProvidersModuleFactory {
...
             const urlObject = new URL(url);
 
             if (!headers.authorization && urlObject.username && urlObject.password) {
                 const authToken = Buffer.from(`${urlObject.username}:${urlObject.password}`).toString('base64');
                 headers.authorization = `Basic ${authToken}`;
             }
 
-            connection = new W3CWebsocket(url, options.protocol, null, headers, null, options.clientConfig);
+            connection = new W3CWebsocket(url, options.protocol, null, headers, requestOptions, options.clientConfig);
```
Comment (nivida) [inline packages/web3-providers/src/factories/ProvidersModuleFactory.js]:
```suggestion
            connection = new W3CWebsocket(url, options.protocol, null, headers, options.requestOptions, options.clientConfig);
```

**Review by nivida — CHANGES_REQUESTED** [review]

**Pull request merged**